mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -688,10 +697,23 @@ public MessageResultIterator getMessages(MessageRange set, FetchGroup fetchGroup
 
     }
 
-    protected Map<MessageUid, MessageMetaData> deleteMarkedInMailbox(final MessageRange range, MailboxSession session) throws MailboxException {
-
+    protected Map<MessageUid, MessageMetaData> deleteWithPreHooks(final MessageRange range, MailboxSession session) throws MailboxException {
         final MessageMapper messageMapper = mapperFactory.getMessageMapper(session);
+        List<MailboxMessage> messagesWillBeDeleted = ImmutableList.copyOf(
+            messageMapper.findInMailbox(mailbox, range, FetchType.Metadata, -1));
```

Comment:
Can you add the information that *-1* means no limit.
Maybe you can introduce a constant for that in that *Limit* class.

Reply:
We should not forget to refactor this once @Arsnael  PR is merged but OK for now.

Reply:
there is no dedicated `Limit` class for message search. But I found `UNLIMITED` constant one